docs/interface.md

```diff
@@ -0,0 +1,56 @@
+# The Interface
+
+All interactions with the Exercism website are handled automatically. Test runners have the single responsibility of taking a solution, running its tests and returning a standardised output.
+
+## Execution
+
+- A test runner should provide an executable script. You can find more information in the [docker.md](https://github.com/exercism/automated-mentoring-support/blob/master/docs/docker.md) file.
+- The script will receive two parameters:
+  - The slug of the exercise (e.g. `two-fer`).
+  - A path to a directory containing the submitted file(s) (with a trailing slash).
+- The script must write a file to that directory named `report.json`
+
+## Output format
+
+The `report.json` file should be structured as followed:
+
+```json
+{
+  "status": "fail",
+  "tests": [
+    {
+      "name": "Test that the thing works",
+      "status": "fail",
+      "expected": "42",
+      "actual": "123123",
+      "message": "Expected 42 but got 123123"
+    }
+  ]
+}
+```
+
+The following overall statuses are valid:
+- `pass`: All tests passed
+- `fail`: At least one test failed
+- `error`: To be used when the tests did not run correctly (including if there are any skips)
```

Comment:
I think that `error` statuses should be an object, not just a string. IMO output like this:

```json
{
  "status": {
    "error": "no tests detected; probable build failure"
  },
  "tests": []
}
```

is much clearer and more understandable than simply:

```json
{
  "status": "error",
  "tests": []
}
```

Reply:
I agree with the need for the error output, but I don't think I want to make the status into an object, as in states without an error message, it feels odd, e.g.

```json
{
  "status": {
    "pass": true
  } 
}

So I think let's add a nullable `error` key instead.

Reply:
I wasn't suggesting changing any of the other statuses. Right now the Rust runner reports

```json
"status": "pass"
```

```json
"status": "fail"
```

```json
"status": {
    "error": "whatever"
}
```

It's a more complicated schema, but it works. A nullable `error` key would also work.